src/rer/ufficiostampa/restapi/services/subscriptions/csv_endpoint.py

```diff
@@ -54,28 +59,58 @@ def reply(self):
                     message=csv_data.get("error", ""),
                 )
             )
+
+        # clone generator for validation checks and processing
+        csv_gen, csv_gen_checks = itertools.tee(csv_data.get("csv", []))
+
+        res = {"errored": []}
+
+        # check for data errors
+        for i, row in enumerate(csv_gen_checks):
+            try:
+                Email().validate(row.get("email", ""))
+            except Exception:
+                msg = translate(
+                    _(
+                        "unvalid_email",
+                        default="[${row}] - row with unvalid email",
+                        mapping={"row": i},
+                    ),
+                    context=self.request,
+                )
+                logger.warning("[ERROR] - {}".format(msg))
+                res["errored"].append(msg)
+
+            try:
+                SendForm.fields["channels"].field.validate(
+                    map(lambda r: r.strip(), row.get("channels").split(","))
+                )
+            except Exception:
+                msg = translate(
+                    _(
+                        "unvalid_channels",
+                        default="[${row}] - row with unvalid channels",
+                        mapping={"row": i},
+                    ),
+                    context=self.request,
+                )
+                logger.warning("[ERROR] - {}".format(msg))
+                res["errored"].append(msg)
+
+        # return if we have errored fields
+        if len(res["errored"]):
+            return res
+
         res = {
```

Comment:
questa definizione di res la metterei in cima con l'altra. Inizializzalo una volta sola, e poi fai return come hai fatto.
Va bene se nel dizionario ci sono anche le altre voci vuote.